pkg/controllers/resources/ingresses/syncer.go

```diff
@@ -117,3 +128,31 @@ func (s *ingressSyncer) SyncToVirtual(ctx *synccontext.SyncContext, event *syncc
 
 	return patcher.CreateVirtualObject(ctx, event.Host, vIngress, s.EventRecorder(), true)
 }
+
+func (s *ingressSyncer) applyLimitByClass(ctx *synccontext.SyncContext, host, virtual *networkingv1.Ingress) bool {
+	//Get the host ingress class and check if it matches the selector
+	if ctx.Config.Sync.FromHost.IngressClasses.Enabled && virtual.Spec.IngressClassName != nil && *virtual.Spec.IngressClassName != "" {
+		pIngressClass := &networkingv1.IngressClass{}
+		err := ctx.PhysicalClient.Get(ctx.Context, types.NamespacedName{Name: *virtual.Spec.IngressClassName}, pIngressClass)
+		if err != nil || pIngressClass.GetDeletionTimestamp() != nil {
+			s.deleteFromHost(ctx, host, virtual, "did not sync ingress %q to host because the ingress class %q couldn't be reached in the host: %s", virtual.GetName(), *virtual.Spec.IngressClassName, err)
+			return true
+		}
+		if !selector.StandardLabelSelectorMatches(pIngressClass, ctx.Config.Sync.FromHost.IngressClasses.Selector) {
+			s.deleteFromHost(ctx, host, virtual, "did not sync ingress %q to host because the ingress class %q in the host does not match the selector under 'sync.fromHost.ingressClasses.selector'", virtual.GetName(), pIngressClass.GetName())
+			return true
+		}
+	}
+	return false
+}
+
+func (s *ingressSyncer) deleteFromHost(ctx *synccontext.SyncContext, host, virtual *networkingv1.Ingress, msgFmt string, args ...any) {
```

Comment:
We basically have this exact same function several times in the PR, can you create DeleteHostObjectWithEvent and move it to where DeleteHostObject is and reuse in the other controllers? I think the only thing that is different is the message on error which we could pass or just use the "Kind" similar to how we do that in the actual DeleteHostObject function, would be probably nice to have for other use cases as well.

Reply:
Sure!